src/utils/yup-locale.ts

```diff
@@ -0,0 +1,94 @@
+import { t } from '@lingui/macro';
+import * as yup from 'yup';
+
+const mixed: Required<yup.LocaleObject['mixed']> = {
+    default: ({ path }) => t`${path} is invalid`,
+    required: ({ path }) => t`${path} is a required field`,
+    defined: ({ path }) => t`${path} must be defined`,
+    notNull: ({ path }) => t`${path} cannot be null`,
+    oneOf: ({ path, values }) => t`${path} must be one of the following values: ${values}`,
+    notOneOf: ({ path, values }) => t`${path} must not be one of the following values: ${values}`,
+    notType: (params) => {
+        const castMsg =
+            params.originalValue != null && params.originalValue !== params.value
+                ? t` (cast from the value \`${yup.printValue(params.originalValue, true)}\`).`
+                : '.';
+        return params.type !== 'mixed'
+            ? t`${params.path} must be a \`${params.type}\` type, ` +
```

Comment:
don't concatenate t strings!
use t`${params.path} must be a \`${params.type}\` type, but the final value was: \`${yup.printValue(params.value, true)}\` ${castMsg}`